### DISCUSSION TITLE
Don't make file extensions obligatory

### DISCUSSION THREAD

**Discussion by duarm** [andmarti1424/sc-im#609] [General]
Hello everyone, thanks for working on sc-im!
I have a bunch of personal tab-separated spreadsheets and some other non conventional formats, since I want to be able to easily read the data without any program. I would like to import that data on sc-im to do some statistics and other things, but sc-im requires a particular extension on my file, otherwise it is "incompatible".  I wouldn't want to just add a "csv" to my file name, since it would be misleading.

I suppose it needs that to know which parser to use, the ideal solution i think would be having a generic parser, in which every information it needs to properly parse the data could be specified as an argument, e.g.:
```shell
sc-im --txt-delim="\ +?\|\|\ +?" --ignore-char="#" --string-delim="\"" --column-delim="\t" my_spreadsheet
```
```csv
1 NAME    STATUS    YEAR    MONTH
2 blabla   || DONE ||2021 || Sep
3 blabla   || DONE ||2021 || Sep
4 # ON_GOING
5 blabla   || ON_GOING || 2021|| Sep
```
Although the regex delimiter would be another request for another time.

But a new option to specify the parser, and only look for the extension if this option is absent, would already solve my problem.
```shell
sc-im --parser csv --txt-delim="\t" my_spreadsheet
```

**Comment by andmarti1424** [comment]
Hello. Yes. I am afraid this is how it currently works. I move this to a discussion, so everyone can give their opinion on this topic. My personal opinion is that it might not really worth modifying bunch of stuff for this.

**Reply by duarm** [reply]
I agree that the generic parser would take longer and not actually necessary, since you can always pre-process the data through piping before sending it to sc-im. But my request, as in the title, is just to "Don't make file extensions obligatory", and the solution would be a '--parser' option, don't think it takes many changes for that. I can do that much and make a pr, if everyone agrees that file extensions shouldn't be necessary.
